engine/src/operators/GroupBy.cpp

```diff
@@ -207,91 +150,82 @@ void distributed_groupby_without_aggregations(
-		return gdf_agg_op_to_groupby_operators(op);
-	});
+   std::vector<cudf::experimental::groupby::aggregation_request> requests;
+   requests.reserve(agg_ops.size());
 
-	cudf::groupby::hash::Options options(false);  // options define null behaviour to be SQL style
+   for(size_t i = 0 ; i < agg_ops.size() ; ++i){
+	   requests.push_back(cudf::experimental::groupby::aggregation_request {.values = aggregation_inputs[i]});
+	   requests[i].aggregations.reserve(agg_ops.size());
 
-	cudf::table group_by_output_table;
-	cudf::table aggrgation_output_table;
-	std::tie(group_by_output_table, aggrgation_output_table) = cudf::groupby::hash::groupby(keys,
-                                            values, ops, options);
-	
-	init_string_category_if_null(group_by_output_table);
-    init_string_category_if_null(aggrgation_output_table);
-
-	group_by_output_columns.resize(group_by_output_table.num_columns());
-	for(size_t i = 0; i < group_by_output_columns.size(); i++) {
-		// TODO percy cudf0.12 port to cudf::column
-//		group_by_output_columns[i].create_gdf_column(group_by_output_table.get_column(i));
-//		group_by_output_columns[i].set_name(group_by_columns[i].name());
-	}
+	   requests[i].aggregations.push_back(std::make_unique<cudf::experimental::aggregation>(agg_ops[i]));
 
-	aggrgation_output_columns.resize(aggrgation_output_table.num_columns());
-	for(size_t i = 0; i < aggrgation_output_columns.size(); i++) {
-		// TODO percy cudf0.12 port to cudf::column
-//		aggrgation_output_columns[i].create_gdf_column(aggrgation_output_table.get_column(i));
-//		aggrgation_output_columns[i].set_name(output_column_names[i]);
-	}
+	   cudf::table_view keys {{group_by_columns}};
+	   cudf::experimental::groupby::groupby obj(keys);
+	   
+	   std::pair<std::unique_ptr<cudf::experimental::table>, std::vector<cudf::experimental::groupby::aggregation_result>> output = obj.aggregate( requests );
+
+	   if (output.first->num_columns()>0) {
+		   std::vector< std::unique_ptr<cudf::column> > gby_cols = output.first->release();
+		   group_by_output_columns.push_back(std::move(gby_cols[0]));
+		   aggregation_output_columns.push_back(std::move(output.second[0].results[0]));
+	   }
+   }
 }
 
-void aggregations_without_groupby(const std::vector<gdf_agg_op> & agg_ops,
-	std::vector<gdf_column_cpp> & aggregation_inputs,
-	std::vector<gdf_column_cpp> & output_columns,
+void aggregations_without_groupby(const std::vector<cudf::experimental::aggregation::Kind> & agg_ops,
+	std::vector<CudfColumnView> & aggregation_inputs,
+	std::vector< std::unique_ptr<cudf::column> > & output_columns,
 	const std::vector<cudf::type_id> & output_types,
 	const std::vector<std::string> & output_column_names) {
 	for(size_t i = 0; i < agg_ops.size(); i++) {
 		switch(agg_ops[i]) {
-		case GDF_SUM:
-		case GDF_MIN:
-		case GDF_MAX:
-			if(aggregation_inputs[i].get_gdf_column()->size() == 0) {
+		case cudf::experimental::aggregation::Kind::SUM:
```

Comment:
is this not implemented it doesnt seem to actually do anything

Reply:
yes it has a TODO there